xmipp

```diff
@@ -385,6 +408,11 @@ def getDefaultBranch(repo):
     # see https://stackoverflow.com/a/32503667/5484355
     # and https://stackoverflow.com/questions/2832269/git-remote-head-is-ambiguous
     # In such a case we return None (and during e.g. clone the client should decide what is the default branch)
+    if repo != XMIPP:
```

Comment:
This part is wrong. Current branch might not be the default branch.

Reply:
Yes it is.

This is what we discussed some days before. If Xmipp repo is in a certain branch, then, the default branch for the rest of the repos is that certain branch (if exists in that repo, if not then the git-defined defaut is).

Note that the `getCurrentBranch('.')` is evaluated in the mail repo.

Reply:
I'm not agree.

Remember that the situation that we want to manage is when having ONLY xmipp repo (very early stages). Then, we want to clone/checkout the branch that is present in xmipp report (if it exists, if not then take the subrepo default want).

In you that commit that you comment, you are taking the current branch in the current repo and that doesn't make sense in that situation.

Reply:
Hi David,

I don't understand what you mean. 

In my implementation, 
if specific branch should be used, it is (https://github.com/I2PC/xmipp/blob/v3.19.04-Apollo-DEVEL/xmipp#L294) ELSE
if the repo exists, its current branch is used (so that we don't change it, https://github.com/I2PC/xmipp/blob/v3.19.04-Apollo-DEVEL/xmipp#L350) ELSE
if the repo (which has not been cloned yet) has the same branch as Xmipp, use that one (https://github.com/I2PC/xmipp/blob/v3.19.04-Apollo-DEVEL/xmipp#L357) ELSE
use repo's default branch (https://github.com/I2PC/xmipp/blob/v3.19.04-Apollo-DEVEL/xmipp#L360)

In your implementation, you're breaking the contract of the 'getDefaultBranch', which does no longer return default branch (as defined by git ), but returns either default branch (https://github.com/I2PC/xmipp/pull/287/files/22bce4e563cf45377fe9ea92e702d10e1847e00a#diff-4e780084a0abd53507bc48714e6c1af1R422) or whatever Xmipp is using at the moment (https://github.com/I2PC/xmipp/pull/287/files/22bce4e563cf45377fe9ea92e702d10e1847e00a#diff-4e780084a0abd53507bc48714e6c1af1R415).

Reply:
For me, that's the correct definition of default branch: the branch that is currently in the main repo if exists here, else the git-default one.

If I want checkout 'someBranch' in xmipp repo, then the default branch for the rest is 'someBranch' but if 'someBranch' is not existing in xmippViz, then default is 'devel'.

I think, this is more sensible.

Reply:
Well, as it is the same and only is regarding the definition of defaultBranch, ok.

I've just uploaded a change in your direction.

If you agree accept the PR and let's merge it. We need to keep pushing.